src/emmy/collection.cljc

```diff
@@ -96,14 +100,9 @@
 ;; structure, or a dataframe-like structure with named fields instead of
 ;; positional fields. Nothing like this exists yet!
 
-#?(:clj
```

Comment:
@littleredcomputer , I want your thoughts here.

I took advantage initially of the ability on the JVM side to extend protocols to java interfaces. That caused trouble when I realized that defrecord instances were picking up all of my implementations...

So here I am backing out the extension to interface in favor of extending these protocols by default only to the concrete map and set implementations.

I've left the vector and seq implementations generic for no good reason other than that I don't know if I've got all of the various seq types listed on the JVM. Also there is no mechanism like defrecord for building a vector-like thing. But still, if we make a deftype that extends IPersistentVector it would accidentally gain all of these implementations.

Is there any reason I should NOT also concretize seq and vector to be safe?

Reply:
I agree with your decision to concretize seq and vector. Rationale: it should be safe for J. Random Thing to implement these interfaces without silently also becoming part of our system. The user can opt another class into the system without requiring anything from us. If someone feels we have overlooked something they can let us know.